src/rules/spaceBeforeFunctionParenRule.ts

```diff
@@ -0,0 +1,167 @@
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+const ALWAYS_OR_NEVER = {
+    enum: ["always", "never"],
+    type: "string",
+};
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: Lint.IRuleMetadata = {
+        description: "Require or disallow a space before function parenthesis",
+        optionExamples: [`[true, {"anonymous": "always", "named": "never", "asyncArrow": "always"}]`],
```

Comment:
seems tedious to need to specify a value for each type of function. How about allowing something like `[true, "always"]` or even `[true]`, which defaults to the same thing.